index.js

```diff
@@ -98,27 +110,31 @@ Consumer.prototype.stop = function () {
   this.stopped = true;
 };
 
-Consumer.prototype._poll = function () {
+Consumer.prototype._poll = function (batchSize) {
+  var consumer = this;
+
   var receiveParams = {
     QueueUrl: this.queueUrl,
     AttributeNames: this.attributeNames,
     MessageAttributeNames: this.messageAttributeNames,
-    MaxNumberOfMessages: this.batchSize,
+    MaxNumberOfMessages: batchSize,
     WaitTimeSeconds: this.waitTimeSeconds,
     VisibilityTimeout: this.visibilityTimeout
   };
 
   if (!this.stopped) {
     debug('Polling for messages');
-    this.sqs.receiveMessage(receiveParams, this._handleSqsResponseBound);
+    this.sqs.receiveMessage(receiveParams, function (err, response) {
+      consumer._handleSqsResponse(err, response, function () {
+        consumer._poll(batchSize);
+      });
+    });
   } else {
     this.emit('stopped');
   }
 };
 
-Consumer.prototype._handleSqsResponse = function (err, response) {
-  var consumer = this;
-
+Consumer.prototype._handleSqsResponse = function (err, response, cb) {
```

Comment:
To retain the value of `batchSize` across calls to `_poll`, I replaced calls `_poll` with a callback to inform the caller that this method has completed handling the sqs response. I find that this makes for cleaner logic in general since `_poll` is now only called when you invoke `start` or by `_poll` itself.